gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1266 @@
+            "The resource named '%s' cannot be deleted because it is not empty", zoneName));
+      }
+    }
+    ProjectContainer projectContainer = projects.get(projectId);
+    ZoneContainer previous = projectContainer.zones.remove(zoneName);
+    return previous == null
+        ? Error.NOT_FOUND.response(String.format(
+        "The 'parameters.managedZone' resource named '%s' does not exist.", zoneName))
+        : new Response(HTTP_NO_CONTENT, "{}");
+  }
+
+  /**
+   * Creates new managed zone and stores it in the collection. Assumes that project exists.
+   */
+  @VisibleForTesting
+  Response createZone(String projectId, ManagedZone zone, String... fields) {
+    // check if the provided data is valid
+    Response errorResponse = checkZone(zone);
+    if (errorResponse != null) {
+      return errorResponse;
+    }
+    // create a copy of the managed zone in order to avoid side effects
+    ManagedZone completeZone = new ManagedZone();
+    completeZone.setName(zone.getName());
+    completeZone.setDnsName(zone.getDnsName());
+    completeZone.setDescription(zone.getDescription());
+    completeZone.setNameServerSet(zone.getNameServerSet());
+    completeZone.setCreationTime(ISODateTimeFormat.dateTime().withZoneUTC()
+        .print(System.currentTimeMillis()));
+    completeZone.setId(BigInteger.valueOf(Math.abs(ID_GENERATOR.nextLong() % Long.MAX_VALUE)));
+    completeZone.setNameServers(randomNameservers());
+    ZoneContainer zoneContainer = new ZoneContainer(completeZone);
+    // create the default NS and SOA records
+    zoneContainer.dnsRecords().set(defaultRecords(completeZone));
+    // place the zone in the data collection
+    ProjectContainer projectContainer = findProject(projectId);
+    ZoneContainer oldValue = projectContainer.zones().putIfAbsent(
+        completeZone.getName(), zoneContainer);
+    if (oldValue != null) {
+      return Error.ALREADY_EXISTS.response(String.format(
+          "The resource 'entity.managedZone' named '%s' already exists", completeZone.getName()));
+    }
+    // now return the desired attributes
+    ManagedZone result = OptionParsers.extractFields(completeZone, fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          String.format("Error when serializing managed zone %s.", result.getName()));
+    }
+  }
+
+  /**
+   * Creates a new change, stores it, and if delayChange > 0, invokes processing in a new thread.
+   */
+  Response createChange(String projectId, String zoneName, Change change, String... fields) {
+    ZoneContainer zoneContainer = findZone(projectId, zoneName);
+    if (zoneContainer == null) {
+      return Error.NOT_FOUND.response(String.format(
+          "The 'parameters.managedZone' resource named %s does not exist.", zoneName));
+    }
+    // check that the change to be applied is valid
```

Comment:
All single line comments can be removed from this method.  The multiple line comment is helpful and should stay though.